engine/src/main/java/org/terasology/world/chunks/Chunks.java

```diff
@@ -0,0 +1,306 @@
+    public static final int SIZE_X = 32;
+    public static final int SIZE_Y = 64;
+    public static final int SIZE_Z = 32;
+
+    public static final int INNER_CHUNK_POS_FILTER_X = Integer.highestOneBit(SIZE_X) - 1;
+    public static final int INNER_CHUNK_POS_FILTER_Y = Integer.highestOneBit(SIZE_Y) - 1;
+    public static final int INNER_CHUNK_POS_FILTER_Z = Integer.highestOneBit(SIZE_Z) - 1;
+
+    public static final int POWER_X = Integer.numberOfTrailingZeros(SIZE_X);
+    public static final int POWER_Y = Integer.numberOfTrailingZeros(SIZE_Y);
+    public static final int POWER_Z = Integer.numberOfTrailingZeros(SIZE_Z);
+
+    public static final byte MAX_LIGHT = 0x0f; // max light for a light source 0-15
+    public static final byte MAX_SUNLIGHT = 0x0f; // max sunlight for sunlight bounded 0-15
+    public static final byte MAX_SUNLIGHT_REGEN = 63;
+    public static final byte SUNLIGHT_REGEN_THRESHOLD = 48;
+
+    public static final Vector3ic CHUNK_POWER = new Vector3i(POWER_X, POWER_Y, POWER_Z);
+    public static final Vector3ic CHUNK_SIZE = new Vector3i(SIZE_X, SIZE_Y, SIZE_Z);
+    public static final Vector3ic INNER_CHUNK_POS_FILTER = new org.joml.Vector3i(INNER_CHUNK_POS_FILTER_X, INNER_CHUNK_POS_FILTER_Y, INNER_CHUNK_POS_FILTER_Z);
+    public static final BlockRegion CHUNK_REGION = BlockRegions.createFromMinAndSize(new Vector3i(), CHUNK_SIZE);
+
+    public static final Vector3ic LOCAL_REGION_EXTENTS = new Vector3i(1, 1, 1);
+
+    private Chunks() {
+    }
+
+    /**
+     * Returns the chunk position of a given coordinate.
+     *
+     * @param val The coordinate of the block
+     * @param chunkPower the size of the chunk in powers of 2
+     * @return The coordinate of the chunk
+     */
+    public static int toChunkPos(int val, int chunkPower) {
+        return (val >> chunkPower);
+    }
+
+    /**
+     * The position of the chunk given the coordinate and size of chunk in powers of 2.
+     * This uses the default power {@link #POWER_X}.
+     *
+     * @param x the x component
+     * @return The coordinate of the chunk
+     */
+    public static int toChunkPosX(int x) {
+        return toChunkPos(x, CHUNK_POWER.x());
+    }
+
+    /**
+     * The position of the chunk given the coordinate and size of chunk in powers of 2.
+     * This uses the default power {@link #POWER_Y}
+     *
+     * @param y the y component
+     * @return The coordinate of the chunk
+     */
+    public static int toChunkPosY(int y) {
+        return toChunkPos(y, CHUNK_POWER.y());
+    }
+
+    /**
+     * The position of the chunk given the coordinate and size of chunk in powers of 2.
+     * This uses the default power {@link #POWER_Z}
+     *
+     * @param z the z component
+     * @return The coordinate of the chunk
+     */
+    public static int toChunkPosZ(int z) {
+        return toChunkPos(z, CHUNK_POWER.z());
+    }
+
+
+    /**
+     * The position of the chunk given the coordinate and size of chunk in powers of 2.
```

Comment:
This method does not take any arguments to set the size of a chunk... Does this method actually do the following:

> returns the position of the chunk that encompasses the given world position